linux/scripts/deb-packaging.sh

```diff
@@ -144,19 +144,39 @@ check_updated_version_number() {
 get_api_version_in_symbols_file() {
   # Extract 1 from "libkeymancore.so.1 libkeymancore #MINVER#"
   local firstline
-  firstline="$(head -1 "debian/${PKG_NAME}.symbols")"
+  local file="$1"
+  firstline="$(head -1 "${file}")"
   firstline="${firstline#"${PKG_NAME}".so.}"
   firstline="${firstline%% *}"
   echo "${firstline}"
 }
 
+get_api_version_from_core() {
+  local api_version file
+  file="$1"
+  api_version=$(cat "${file}")
+  # Extract major version number from "1.0.0"
+  api_version=${api_version%%.*}
+  echo "${api_version}"
+}
+
+# Check if the API version got updated
+# Returns:
+#   0 - if the API version got updated
+#   1 - the .symbols file got changed but the API version didn't get updated
+#   2 - if we're in the alpha tier and the API version got updated since
+#       the last stable version
+# NOTE: it is up to the caller to check if this is a major version
+# change that requires an API version update.
 is_api_version_updated() {
-  local NEW_VERSION OLD_VERSION TIER
-  git checkout "${GIT_BASE}" -- "debian/${PKG_NAME}.symbols"
-  OLD_VERSION=$(get_api_version_in_symbols_file)
-  git checkout "${GIT_SHA}" -- "debian/${PKG_NAME}.symbols"
-  NEW_VERSION=$(get_api_version_in_symbols_file)
-  if (( NEW_VERSION > OLD_VERSION )); then
+  local OLD_API_VERSION NEW_API_VERSION  TIER TMPFILE
+  TMPFILE=$(mktemp)
+  git cat-file blob "${GIT_BASE}:linux/debian/${PKG_NAME}.symbols" > "${TMPFILE}"
+  OLD_API_VERSION=$(get_api_version_in_symbols_file "${TMPFILE}")
```

Comment:
This is looking better but there are a lot of repeated file and git references.

I think it would be good if we could make another wrapper function that does the full cat-file and parse here, and then that function would also manage the lifecycle of `$TMPFILE`.

```bash
  local SYMBOL_FILE="linux/debian/${PKG_NAME}.symbols"
  local GIT_STABLE_BRANCH="origin/stable-${STABLE_VERSION}.0"
  local OLD_API_VERSION=$(get_api_version_from_git_blob_symbols_file "${GIT_BASE}:${SYMBOL_FILE}")
  if (( OLD_API_VERSION == -1 )); then
    # error occurred, ...
  fi

  ...
  
  local STABLE_API_VERSION=$(get_api_version_from_git_blob_symbols_file "${GIT_STABLE_BRANCH}:${SYMBOL_FILE}"
  if (( STABLE_API_VERSION == -1 )); then
    STABLE_API_VERSION=$(get_api_version_from_git_blob_core "${GIT_STABLE_BRANCH}:core/CORE_API_VERSION.md")
    if (( STABLE_API_VERSION == -1 )); then
      ...
    fi
  fi
```